pkg/agent/core/state.go

```diff
@@ -1061,19 +1178,29 @@ func (h MonitorHandle) StartingDownscaleRequest(now time.Time, resources api.Res
 	h.s.Monitor.DownscaleFailureAt = nil
 }
 
-func (h MonitorHandle) DownscaleRequestAllowed(now time.Time) {
+func (h MonitorHandle) DownscaleRequestAllowed(now time.Time, rev vmv1.RevisionWithTime) {
 	h.s.Monitor.Approved = &h.s.Monitor.OngoingRequest.Requested
 	h.s.Monitor.OngoingRequest = nil
+	revsource.Propagate(now,
+		rev,
+		&h.s.Monitor.CurrentRevision,
+		h.s.Config.ObservabilityCallbacks.MonitorLatency,
+	)
 }
 
 // Downscale request was successful but the monitor denied our request.
-func (h MonitorHandle) DownscaleRequestDenied(now time.Time) {
+func (h MonitorHandle) DownscaleRequestDenied(now time.Time, targetRevision vmv1.RevisionWithTime) {
 	h.s.Monitor.DeniedDownscale = &deniedDownscale{
 		At:        now,
 		Current:   *h.s.Monitor.Approved,
 		Requested: h.s.Monitor.OngoingRequest.Requested,
 	}
 	h.s.Monitor.OngoingRequest = nil
+	revsource.Propagate(now,
+		targetRevision,
+		&h.s.Monitor.CurrentRevision,
+		h.s.Config.ObservabilityCallbacks.MonitorLatency,
+	)
```

Comment:
Similar question here as with the scheduler - what happens when downscale is denied?

Reply:
The propagation doesn't happen -> we don't measure the latency.

Reply:
Here we still measure the latency for the vm-monitor even though it was denied, right? Is that intentional? (if so: what are the expected semantics for component latency?)

Reply:
> Well, the distribution of latency for successful requests

My current understanding is either:

* Component latency should only look at individual request latency
* Component latency should only be related end-to-end scaling

If it's the first one, then presumably we shouldn't be using `revsource` for this (we'd just want a simple histogram metric looking at the time difference since we started the request, right?).

If it's the second one, then we should treat denial as failure, because that doesn't get us closer to scaling.

Does that make sense?

Reply:
It is "Component latency should only look at individual request latency".

The implementation will remain as-is for now, and later can be simplified.

I should check the metric name, that it is clearly expresses semantics.